exercises/square-root/instructions.md

```diff
@@ -0,0 +1,18 @@
+# Instructions
+
+Your task is to implement a calculation of the square root of a given number.
+
+- Try to avoid using the pre-existing math libraries of your language.
+- As input you’ll be given a positive whole number, i.e. 1, 2, 3, 4…
+- You also only need to output a positive whole number.
```

Comment:
```suggestion
- You are only required to handle cases where the output is a positive whole number.
```

Reply:
The suggested change is misleading because it seems to preclude approximating the output to a positive whole number even when it is not the exact result, which most implementations of integer square root will do.  It biases the student to implementations that only handle perfect squares, which overly narrows the exercise.

Reply:
I think this is a good suggestion, we can see what the requirements are from the tests.  What is allowed will pass the tests, even (and partly because of) the tests not being exhaustive.

The tests guide what the requirement is, and potentially directly what is not allowed.

The reading can be a bit ambiguous, but may  not contradict.

**What about leaving that line out?**  We can see what the tests require, and the tests are going to communicate, and we also encourage the students to read them.

Also, "output" does not happen in the sense of "to  a screen or file" but what results is a positive whole number.

The tests are also a source of bias, though, we may not be able to avoid that.

Reply:
OK, I've replaced "output" with "return".

Reply:
I second IsaacG's suggestion. It emphasizes that we are limited to inputs that yield a specific type of result. The current change just focuses on what the result is. The difference is subtle but important.